### ISSUE TITLE
clojars archive is outdated & has incremented version

### ISSUE THREAD

**Issue by moquist** [swannodette/om-async-tut#6]
The latest archive in clojars is for this commit: https://github.com/swannodette/om-async-tut/commit/0e0386176af12e28df3ab8051b2ece112180742d

Unfortunately, that commit has some bugs. Those bugs have been fixed in master but the fixes aren't pushed to clojars.

In addition, the clojars version is 0.1.9, whereas master is version 0.1.8.

**Comment by moquist** [comment]
Maybe the archive in clojars is intentionally behind master because https://github.com/omcljs/om/wiki/Intermediate-Tutorial will require revision.

If so, I'm happy to help with that.

Is the goal of the tutorial revision to end up with https://github.com/bensu/om-intermediate-tut instead? (For example, not using goog.dom just for goog.dom/getElement?)

**Comment by bensu** [comment]
Thanks! I just published the updated version of the template under [om-intermediate-template](https://clojars.org/om-intermediate-template/lein-template) and updated the wiki accordingly. I read it once for typos and tried the template locally but that is not a guarantee it will work. Would you mind trying the new template with the updated wiki?

**Comment by swannodette** [comment]
Once it's been proofed I'll happily push to Clojars. Thanks all.

**Comment by moquist** [comment]
I'm proofing it now and will let you know when I'm done.

**Comment by moquist** [comment]
Just to clarify -- after @bensu's changes the wiki refers to a different template name, so this repo is the historical parent but is no longer directly relevant to the tutorial. I'm going through https://github.com/omcljs/om/wiki/Intermediate-Tutorial as it's currently written, using om-intermediate-template.

**Comment by bensu** [comment]
Yes. I pointed the wiki to the updated template and published it so somebody else could also test it. As soon as the changes are proofed and David publishes them to Clojars, I'll restore the wiki towards this version, the canonical one. Thanks

**Comment by moquist** [comment]
@bensu: cool. This is a good step forward.

I proofed and it works for me, with one caveat: the 500 we get back from the server after the last step isn't actually coming from the 'create-class handler! This isn't a regression, because the previous version of the tutorial had the same issue. I haven't had time to figure it out yet, though. Here's the stack trace; maybe the issue will be obvious to you: https://gist.github.com/moquist/b648fe2797e70797448d

@bensu: I removed a few more vestigial snippets throughout the tutorial and will send a pull request shortly for the final code state as I have it. None of them are crucial, but I think they demonstrate better refactoring.

**Comment by bensu** [comment]
That's great! For completeness, I'll add here that the final state of the code is at [om-intermediate-tut](https://github.com/bensu/om-intermediate-tut) 

I hadn't realized the POST error was not what it was supposed to be. Since I am planning to learn more about om-sync. I'll look at the stack trace then and try to figure it (I can't at a glance).

**Comment by moquist** [comment]
Yep, pull request sent for https://github.com/bensu/om-intermediate-tut .

I'm planning to look at the POST error later today, but please do let me know if you figure it out already. :)

**Comment by moquist** [comment]
@bensu: Figured the stack-trace out. wrap-edn-params assumes the EDN in the body is a map, and om-sync is sending the raw collection, which in this case is a vector of maps. Here's the offending line: https://github.com/tailrecursion/ring-edn/blob/master/src/ring/middleware/edn.clj#L33

Here's a tiny reproduction of an identical scenario:
=> (merge {:a "b"} [:c])

I think two things should probably happen:
1. this README (https://github.com/tailrecursion/ring-edn) should clarify that wrap-edn-params requires the EDN body to be a map (this is implicit in the name "params"), and
2. the tutorial should ditch wrap-edn-params after the class creation stuff is in place and just parse the EDN body with our own tiny middleware.

What do you guys think?

**Comment by moquist** [comment]
This demonstrates what I suggested above: https://github.com/moquist/om-intermediate-tut/compare/bensu:master...master?diff=split&name=master

Writing even a small EDN-body-parsing middleware does seem like a distraction for the tutorial. Maybe there's some other lib than ring.middleware.edn that already handles this for us. OTOH, it _does_ give the tutee some pretty obvious clues about how to see exactly what om-sync is sending, and how they can get their hands on the data -- and that seems worthwhile.

If you guys want to go this way, I'm happy to send a pull request and update the wiki.

**Comment by bensu** [comment]
I agree with both points. I don't think adding the middleware would be a big distraction since the tutorial only skims through the back end code. Adding the middleware just means changing a paragraph to that narrative. 

I tried it and it works as expected :) I'll be glad to update the template and publish it after you send me the pull request to `om-intermediate-tut`.

**Comment by moquist** [comment]
@bensu: Thanks!

**Comment by bensu** [comment]
Changes added to the template, to the wiki, and deployed to Clojars. Thanks for the help!

**Comment by moquist** [comment]
Likewise.
